src/io/flutter/editor/NativeEditorNotificationProvider.java

```diff
@@ -0,0 +1,138 @@
+  public EditorNotificationPanel createNotificationPanel(@NotNull VirtualFile file, @NotNull FileEditor fileEditor) {
+    if (!file.isInLocalFileSystem()) {
+      return null;
+    }
+    final VirtualFile root = findRootDir(file);
+    if (root == null) {
+      return null;
+    }
+
+    final String actionName;
+    if (root.getName().equals("android")) {
+      actionName = "flutter.androidstudio.open";
+    }
+    else if (root.getName().equals("ios")) {
+      actionName = "flutter.xcode.open";
+    }
+    else {
+      return null;
+    }
+
+    NativeEditorActionsPanel panel = new NativeEditorActionsPanel(file, root, actionName);
+    return panel.isValidForFile() ? panel : null;
+  }
+
+  private VirtualFile findRootDir(@NotNull VirtualFile file) {
+    // Return the top-most parent of file that is a child of the project directory.
+    final VirtualFile projectDir = myProject.getBaseDir();
+    VirtualFile parent = file.getParent();
+    if (projectDir.equals(parent)) {
+      return null;
+    }
+    VirtualFile root = parent;
+    while (parent != null) {
+      parent = parent.getParent();
+      if (projectDir.equals(parent)) {
+        return root;
+      }
+      root = parent;
+    }
+    return null;
+  }
+
+  class NativeEditorActionsPanel extends EditorNotificationPanel {
+    final VirtualFile myFile;
+    final VirtualFile myRoot;
+    final AnAction myAction;
+
+    NativeEditorActionsPanel(VirtualFile file, VirtualFile root, String actionName) {
+      super(EditorColors.GUTTER_BACKGROUND);
+      myFile = file;
+      myRoot = root;
+      myAction = ActionManager.getInstance().getAction(actionName);
+
+      icon(FlutterIcons.Flutter);
+      text("Flutter commands");
+
+      final Presentation present = myAction.getTemplatePresentation();
+      final HyperlinkLabel label = createActionLabel(present.getText(), this::performAction);
+      label.setToolTipText(present.getDescription());
+    }
+
+    private boolean isValidForFile() {
+      final DataContext context = makeContext();
+      final AnActionEvent event = AnActionEvent.createFromDataContext(ActionPlaces.EDITOR_TOOLBAR, null, context);
+      // Ensure this project is a Flutter project by updating the menu action. It will only be visible for Flutter projects.
+      myAction.update(event);
+      if (event.getPresentation().isVisible()) {
+        // The menu items are visible for certain elements outside the module directories.
+        return FileUtil.isAncestor(myRoot.getPath(), myFile.getPath(), true);
+      }
+      return false;
+    }
+
+    private void performAction() {
+      final Presentation present = myAction.getTemplatePresentation();
+      final DataContext context = makeContext();
+      final AnActionEvent event = AnActionEvent.createFromDataContext(ActionPlaces.EDITOR_TOOLBAR, present, context);
+      // Open Xcode or Android Studio. If already running AS then just open a new window.
+      myAction.actionPerformed(event);
```

Comment:
Here we'll want to update the actions so that analytics can track how they were performed. That is, we really want to know how often the actions were triggered from the banner (vs. context menu).